test/functional/legacy/delete_spec.lua

```diff
@@ -52,6 +52,10 @@ describe('Test for delete()', function()
         silent !ln -s Xfile Xlink
       endif
     ]])
+    if eval('v:shell_error') == 1 then
+      eq(0, eval("delete('Xfile')"))
```

Comment:
Probably should just do `os.remove('Xfile')` in `before_each`, since this isn't a useful test.